src/org/opendatakit/briefcase/operations/Export.java

```diff
@@ -22,157 +22,105 @@
-  private static final Param<Date> END = Param.arg("end", "export_end_date", "Export end date", Export::toDate);
+  private static final Param<LocalDate> START = Param.arg("start", "export_start_date", "Export start date", LocalDate::parse);
+  private static final Param<LocalDate> END = Param.arg("end", "export_end_date", "Export end date", LocalDate::parse);
   private static final Param<Void> EXCLUDE_MEDIA = Param.flag("em", "exclude_media_export", "Exclude media in export");
   private static final Param<Void> OVERWRITE = Param.flag("oc", "overwrite_csv_export", "Overwrite files during export");
-  private static final Param<String> PEM_FILE = Param.arg("pf", "pem_file", "PEM file for form decryption");
-
-  public static Date toDate(String s) {
-    try {
-      return DATE_FORMAT.parse(s);
-    } catch (ParseException e) {
-      LOGGER.error("bad date range", e);
-      return null;
-    }
-  }
-
+  private static final Param<Path> PEM_FILE = Param.arg("pf", "pem_file", "PEM file for form decryption", Paths::get);
 
   public static Operation EXPORT_FORM = Operation.of(
       EXPORT,
       args -> export(
           args.get(STORAGE_DIR),
           args.get(FORM_ID),
-          args.get(FILE),
           args.get(EXPORT_DIR),
-          args.getOrNull(START),
-          args.getOrNull(END),
+          args.get(FILE),
           !args.has(EXCLUDE_MEDIA),
           args.has(OVERWRITE),
+          args.getOptional(START),
+          args.getOptional(END),
           args.getOptional(PEM_FILE)
       ),
       Arrays.asList(STORAGE_DIR, FORM_ID, FILE, EXPORT_DIR),
       Arrays.asList(PEM_FILE, EXCLUDE_MEDIA, OVERWRITE, START, END)
   );
 
-  public static void export(String storageDir, String formid, String fileName, String exportPath, Date startDateString, Date endDateString, boolean exportMedia, boolean overwrite, Optional<String> pemKeyFile) {
+  public static void export(String storageDir, String formid, Path exportPath, String baseFilename, boolean includeMediaFiles, boolean overwriteFiles, Optional<LocalDate> startDate, Optional<LocalDate> endDate, Optional<Path> pemFile) {
+    CliEventsCompanion.attach(log);
     bootCache(storageDir);
-    BriefcaseFormDefinition formDefinition = null;
-    List<BriefcaseFormDefinition> forms = FileSystemUtils.getBriefcaseFormList();
-    for (int i = 0; i < forms.size(); i++) {
-      BriefcaseFormDefinition x = forms.get(i);
-      if (formid.equals(x.getFormId())) {
-        formDefinition = x;
-        break;
-      }
-    }
+    Optional<BriefcaseFormDefinition> maybeFormDefinition = FileSystemUtils.getBriefcaseFormList().stream()
+        .filter(form -> form.getFormId().equals(formid))
+        .findFirst();
 
-    if (formDefinition == null) {
-      LOGGER.error("Form not found");
-      return;
-    }
+    if (!maybeFormDefinition.isPresent())
+      throw new FormNotFoundException(formid);
+
+    BriefcaseFormDefinition formDefinition = maybeFormDefinition.get();
```

Comment:
I’ve learned the painful lesson (in Scala)—several times—never to call get on an Option. Even when I could prove it’s non-empty. I’m new to Java’s implementation of these functional constructs, but I think this guideline might apply here, too. What do you think of the following change?

![never call get](https://user-images.githubusercontent.com/55099/36115006-878d1f50-0fe6-11e8-939d-0a1f3fbeec33.png)

Reply:
Wait—something better coming

Reply:
I just discovered Optional.orElseThrow. Perfect for this use, yes?

Reply:
Nice catch! Yep, some commits before that, I needed to make a side effect (sending an event) that I wouldn't be ok to do on an `orElseThrow` but then I removed the side effect and forgot about that block. Thanks!

Reply:
I was very pleased to learn from the investigation! It’s great to be learning with you.